### PR TITLE
Add prepend_javascript_pack_tag helper

### PR DESCRIPTION
**Pull request by paypro-leon**
In our codebase we had a similar approach for Webpacker 5 for `append_javascript_pack_tag` helper, but we used `unshift` instead of appending to the array.

Using `append_javascript_pack_tag` causes issues for us where we expect some entries to go before the appended entries added by the main view or partials.

Adding `prepend_javascript_pack_tag` to allow a pack to go to the beginning of the entry array would give us more flexibility in the order of the entries inserted by `javascript_pack_tag`.

### CODE REVIEW

**Review by ahangarha — APPROVED** [review]
Thanks for your contribution.
And thanks for updating the changelog and adding specs for these changes.

**Comment by paypro-leon** [comment]
@ahangarha No problem, and thank you guys for continuing and improving `webpacker`!

File: lib/webpacker/helper.rb [hunk 1]
```diff
@@ -180,20 +180,31 @@ def append_stylesheet_pack_tag(*names)
...
 
   private
 
+    def update_javascript_pack_tag_queue(defer:)
+      if @javascript_pack_tag_loaded
+        raise "You can only call #{caller_locations(1..1).first.label} before javascript_pack_tag helper. " \
+        "Please refer to https://github.com/shakacode/shakapacker/blob/master/README.md#view-helper-append_javascript_pack_tag-and-append_stylesheet_pack_tag for the usage guide"
+      end
+
+      yield defer ? :deferred : :non_deferred
```
Comment (justin808) [inline lib/webpacker/helper.rb:202]: I prefer adding the argument with parens.

```rb
yield(defer ? :deferred : :non_deferred)
```

File: lib/webpacker/helper.rb [hunk 2]
```diff
@@ -180,20 +180,31 @@ def append_stylesheet_pack_tag(*names)
...
+  end
 
-    hash_key = defer ? :deferred : :non_deferred
-    javascript_pack_tag_queue[hash_key] |= names
-
-    # prevent rendering Array#to_s representation when used with <%= … %> syntax
-    nil
+  def prepend_javascript_pack_tag(*names, defer: true)
+    update_javascript_pack_tag_queue(defer: defer) do |hash_key|
+      javascript_pack_tag_queue[hash_key].unshift *names
```
Comment (justin808) [inline lib/webpacker/helper.rb:190]: Use parens for argument.

**Review by justin808 — CHANGES_REQUESTED** [review]
Looking great! I'd like to ship this ASAP.

We're mainly missing the documentation. Please check for the docs for `append_javascript_pack_tag` and update accordingly.

**Comment by paypro-leon** [comment]
@justin808 Wasn't sure if we should split up the `append_javascript_pack_tag` and `prepend_javascript_pack_tag` documentation in the README. For now I kept them together.

**Review by vaukalak — APPROVED** [review]

**Comment by justin808** [comment]
Big thanks @paypro-leon!

**Pull request merged**